blueflood-core/src/test/java/com/rackspacecloud/blueflood/service/BluefloodServiceStarterTest.java

```diff
@@ -0,0 +1,191 @@
+package com.rackspacecloud.blueflood.service;
+
+import static junit.framework.Assert.*;
+
+import org.junit.*;
+
+import java.io.IOException;
+
+public class BluefloodServiceStarterTest {
+
+    @Before
+    public void setUp() throws IOException {
+
+        Configuration config = Configuration.getInstance();
+
+        config.init();
+
+        config.setProperty(CoreConfig.ZOOKEEPER_CLUSTER, "NONE");
+
+        config.setProperty(CoreConfig.INGEST_MODE, "false");
+        config.setProperty(CoreConfig.ROLLUP_MODE, "false");
+        config.setProperty(CoreConfig.QUERY_MODE, "false");
+
+        BluefloodServiceStarter.shouldInstantiateRestartGauge = false;
+    }
+
+    @Test
+    public void testAllModesDisabled() {
+
+        // given
+        Configuration config = Configuration.getInstance();
+        config.setProperty(CoreConfig.INGEST_MODE, "false");
+        config.setProperty(CoreConfig.QUERY_MODE, "false");
+        config.setProperty(CoreConfig.ROLLUP_MODE, "false");
+
+        // when
+        BluefloodServiceStarter.run();
```

Comment:
That is the opposite of [what @shintasmith suggested](https://github.com/rackerlabs/blueflood/pull/609#discussion_r51589607). I think it's ok for now to leave it as is.